x/valset/keeper/keeper.go

```diff
@@ -525,16 +528,26 @@ func (k Keeper) Jail(ctx sdk.Context, valAddr sdk.ValAddress, reason string) err
 	if val.IsJailed() {
 		return ErrValidatorAlreadyJailed.Format(valAddr.String())
 	}
+
+	consensusPower := val.GetConsensusPower(k.powerReduction)
+	totalConsensusPower := int64(0)
 	count := 0
 	k.staking.IterateValidators(ctx, func(_ int64, val stakingtypes.ValidatorI) bool {
 		if val.IsBonded() && !val.IsJailed() {
+			totalConsensusPower += val.GetConsensusPower(k.powerReduction)
 			count++
 		}
 		return false
 	})
+
 	if count == 1 {
 		return ErrCannotJailValidator.Format(valAddr).WrapS("number of active validators would be zero then")
 	}
+
+	if float64(consensusPower)/float64(totalConsensusPower) > cJailingNetworkShareProtection {
```

Comment:
Oh, it's done per-validator?  We could still hit a case where 2 validators are jailed at the same time to bring us below consensus.  Would it be more desirable to prevent the entire jailing event if we're about to jail validators summing 25% of consensus?

Reply:
`totalConsensusPower` is recalculated for every jail event. 

In case we have 5 validators, each with a stake of `20`, and three of them are suddenly without a Pigeon and should be jailed. What would happen is:

Validator `A` holds 20 stakes (out of 100). It's below 25%. Validator is jailed.
Validator `B` holds 20 stakes (out of 80). That's exactly 25%. Validator is jailed.
Validator `C` and the others holds 20 stakes (out of 60). That's above 25%. Validator is not jailed.

Reply:
That makes total sense and should protect us then from jailing into a chain halt 👍